interp/src/primitives.rs

```diff
@@ -672,6 +680,20 @@ impl StdMemD2 {
             update: None,
         }
     }
+
+    pub fn initialize_memory(&mut self, vals: &[Value]) {
+        assert_eq!((self.d0_size * self.d1_size) as usize, vals.len());
+
+        for (idx, val) in vals.iter().enumerate() {
+            assert_eq!(val.len(), self.width as usize);
+            self.data[idx] = val.clone()
+        }
+    }
+
+    #[inline]
+    fn calc_addr(&self, addr0: u64, addr1: u64) -> u64 {
+        addr1 + addr0 * self.d1_size
```

Comment:
LGTM! Super trivial comment here, and feel free to ignore, but I usually see this kind of calculation written the other way: `addr0 * self.d1_size + addr1`. It sort of emphasizes that, as you walk down the array, the rightmost thing changes the fastest.